dockstore/dockstore-ui2
# Add AI topic approval bubble if user selects the AI topic option

**Description**
Build will fail until https://github.com/dockstore/dockstore/pull/5949 is merged (non-breaking change). Currently a draft PR until the webservice PR is merged and the builds pass.

This PR modifies the AI topic presentation so that we can track if a user manually selected the AI topic (and thus approves of it).

In the topic editing dialog:
- An info card is displayed if the AI topic was automatically selected (because the entry had no other topics) when we updated the AI topic.
- When the user selects the AI topic, a confirmation message is displayed to ensure that the user has reviewed it. 
  - This confirmation UI is inspired by the Request DOI dialog that asks the user to confirm that they want to snapshot the version.
- Once the AI topic selection is saved, the AI topic card in the topic editing dialog will show an Approved bubble.

In the public display of AI topics:
- If the topic was approved by the user, the AI bubble is **not** displayed.
- Note that in the private view, the AI bubble is always displayed so the user knows what the topic selection is in the Workflow Info card.


**Screenshots**

<details>
  <summary>Workflow with an automatically selected AI topic because it has no other topics</summary>


- Upon opening the dialog, there is an info card since AI topic was automatically selected and not yet approved by the user.
  ![Screenshot 2024-07-25 at 13-32-32 Dockstore My Workflows](https://github.com/user-attachments/assets/0e0ebdc3-943a-4d3d-ae22-af0fbd6e0036)

- Clicking Save prompts the user to confirm that they have reviewed the topic.
  ![Screenshot 2024-07-25 at 13-32-41 Dockstore My Workflows](https://github.com/user-attachments/assets/30844af5-e680-419b-a387-2e0597d1bdc5)

- After confirming, the AI topic is displayed with an AI bubble in the private view of the workflow. The AI bubble is there to indicate the topic type.
  ![Screenshot 2024-07-25 at 13-32-56 Dockstore My Workflows](https://github.com/user-attachments/assets/3c7b872c-b56c-4b19-b34b-8eae72ce81bb)

- Opening the dialog again, the AI topic has an Approved bubble
  ![Screenshot 2024-07-25 at 13-33-05 Dockstore My Workflows](https://github.com/user-attachments/assets/c84835f2-8bb4-468f-af0f-91aca42d103d)

- In the public view of the workflow, no AI bubble is displayed
  ![Screenshot 2024-07-25 at 13-33-32 Dockstore Workflow github com_kathy-t_test-authors_foobar github com_kathy-t_test-authors_foobar](https://github.com/user-attachments/assets/d1d89b4c-baeb-4c7d-9654-9be53c5e66b8)

- On the search page, no AI bubble is displayed
  ![Screenshot 2024-07-25 at 13-34-27 Dockstore Workflow](https://github.com/user-attachments/assets/df0bf263-e39e-4a30-a8e2-c94d39164cd0)

</details>

<details>
<summary>Workflow has an AI topic but it wasn't automatically selected</summary>

- There is no info card displayed for the AI topic. Debated whether there should be something that indicates that the topic has not been reviewed yet (a Not Reviewed bubble?) but decided against it because it doesn't seem that meaningful. For example, the user may have reviewed it and they still prefer their other topic.
  ![Screenshot 2024-07-25 at 13-40-13 Dockstore My Workflows](https://github.com/user-attachments/assets/cf78e2e9-07e4-4720-86f5-1afa5868a5f5)

- Clicking Save prompts the user to confirm that they have reviewed the topic.
![Screenshot 2024-07-25 at 13-40-23 Dockstore My Workflows](https://github.com/user-attachments/assets/44e9d58a-3a78-402d-a0e4-3e6ccab0b7d0)

- Once it's saved and the user re-opens the dialog, the AI topic has an Approved bubble
  ![Screenshot 2024-07-25 at 13-40-30 Dockstore My Workflows](https://github.com/user-attachments/assets/2a2b2353-387d-4db5-9720-75885030bef9)

</details>

<details>
<summary>UI styling changes to Request DOI confirmation message</summary>

- I updated the styling of the Request DOI confirmation message to match the confirmation message for this PR.

- Before
  ![image](https://github.com/user-attachments/assets/75f4be05-9bc3-4f2b-b09d-ede50c6419fb)

- After
  ![Screenshot 2024-07-25 at 11-07-13 Dockstore My Workflows](https://github.com/user-attachments/assets/de711b40-77f0-456d-8743-1f218a07fd77)


</details>

**Review Instructions**
1. Use this [Swagger endpoint](https://qa.dockstore.org/api/static/swagger-ui/index.html#/extendedGA4GH/updateAITopic) to update a public workflow that you own with an AI topic. You must be an admin or curator to use this endpoint. This workflow should have **no** other types of topics so that the AI selection is automatically set to AI.
2. Go to Dockstore and navigate to My Workflows > the workflow from step 1. Click the Edit button for topics.
3. Verify that there is an info card for the AI topic option.
4. Click Save (ensuring that AI is still selected) and verify that there is a confirmation message.
5. Once it's saved, verify that the Topic display in the My Workflows view has an AI bubble.
6. Navigate to the public view of the workflow and verify that it does **NOT** have an AI bubble.
7. Find the workflow in the search page and verify that it does **NOT** have an AI bubble.
8. Repeat step 1 for a public workflow that already has a topic.
9. Go to Dockstore and navigate to My Workflows > the workflow from step 8. Click the Edit button for topics.
10. Verify that there is **NO** info card for the AI topic option because it was not automatically selected.
11. Repeat steps 4-7 for this workflow.

**Issue**
https://github.com/dockstore/dockstore/issues/5866
https://ucsc-cgl.atlassian.net/browse/DOCK-2521

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!
